src/serverrpc.cpp

```diff
@@ -0,0 +1,137 @@
+            { "enabled", pServer->GetRecordingEnabled() },
+            { "recordingDirectory", pServer->GetRecordingDir() },
+        };
+
+        // create result object
+        QJsonObject result{
+            { "name", pServer->GetServerName() },
+            { "city", pServer->GetServerCity() },
+            { "countryId", pServer->GetServerCountry() },
+            { "welcomeMessage", pServer->GetWelcomeMessage() },
+            { "registrationStatus", pServer->GetSvrRegStatus() },
+            { "clients", clients },
+            { "recorder", recorder },
+        };
+        response["result"] = result;
+        Q_UNUSED ( params );
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/setServerName", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        auto jsonServerName = params["serverName"];
+        if ( !jsonServerName.isString() )
+        {
+            response["error"] = CRpcServer::CreateJsonRpcError ( -32602, "Invalid params: serverName is not a string" );
+            return;
+        }
+
+        pServer->SetServerName ( jsonServerName.toString() );
+        response["result"] = "ok";
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/setWelcomeMessage", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        auto jsonWelcomeMessage = params["welcomeMessage"];
+        if ( !jsonWelcomeMessage.isString() )
+        {
+            response["error"] = CRpcServer::CreateJsonRpcError ( -32602, "Invalid params: welcomeMessage is not a string" );
+            return;
+        }
+
+        pServer->SetWelcomeMessage ( jsonWelcomeMessage.toString() );
+        response["result"] = "ok";
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/setRecordingDirectory", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        auto jsonRecordingDirectory = params["recordingDirectory"];
+        if ( !jsonRecordingDirectory.isString() )
+        {
+            response["error"] = CRpcServer::CreateJsonRpcError ( -32602, "Invalid params: jsonRecordingDirectory is not a string" );
+            return;
+        }
+
+        pServer->SetRecordingDir ( jsonRecordingDirectory.toString() );
+        response["result"] = "acknowledged";
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/startRecording", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        pServer->SetEnableRecording ( true );
+        response["result"] = "acknowledged";
+        Q_UNUSED ( params );
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/stopRecording", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        pServer->SetEnableRecording ( false );
+        response["result"] = "acknowledged";
+        Q_UNUSED ( params );
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/restartRecording", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        pServer->RequestNewRecording();
+        response["result"] = "acknowledged";
```

Comment:
Is there a reason for using `acknowledged` here instead of `ok` like in `setWelcomeMessage`?

Reply:
@hoffie If the recording cannot be enabled (possibly due to non-existent parent directory, permissions error, or some other reason), right now it seems that the results are delivered asynchronously, so you cannot be sure that the method succeeded, so I am very reluctant to return "ok".

After calling these methods return "acknowledged", a client should check the status by calling (or polling) `jamulusserver/getServerStatus` afterwards. I know this is not ideal but I want to avoid adding any more complexity into existing code. I hope this clarifies.